aten/src/ATen/native_parse.py

```diff
@@ -88,11 +98,35 @@ def parse_arguments(args, func_decl, func_name, func_return):
         # be better if we just named everything and matched by name.
         if is_out_fn and arg_idx < len(func_return):
             argument_dict['output'] = True
+        argument_dict["type"], argument_dict["annotation"] = get_type_annotation(argument_dict["type"])
+        # We assume that type annotations are only used for output tensors
```

Comment:
All of this also depends on how clean we want this to be, if we're operating under the assumption that eventually the native_parser only consists of forwarding the function declarations so that they're parsed by the JIT and Autograd directly. Maybe it's better to force functions that use "advanced" syntax such as annotations to comply exactly with the JIT IR.